src/vm/parser.rs

```diff
@@ -236,21 +241,27 @@ pub fn parse_lexed(mut input: Vec<(LexItem, u32)>) -> Result<Vec<SymbolicExpress
                 }
             },
             LexItem::Variable(value) => {
+                let end_column = column_pos + (value.len() as u32) - 1;
                 let mut expression = SymbolicExpression::atom(value);
-                expression.set_line_number(line_number);
+                expression.set_span(line_pos, column_pos, line_pos, end_column);
 
                 match parse_stack.last_mut() {
                     None => output_list.push(expression),
-                    Some((ref mut list, _)) => list.push(expression)
+                    Some((ref mut list, _, _)) => list.push(expression)
                 };
             },
-            LexItem::LiteralValue(value) => {
+            LexItem::LiteralValue(length, value) => {
+                let mut end_column = column_pos + (length as u32);
+                // Avoid underflows on cases like empty strings
```

Comment:
hmm, this why spans are usually represented non-inclusive on the end index, i.e., they represent the mathematical span `[start_index, end_index)` -- I'm not hugely concerned about this, but it would simplify some of this kind of code. Do diagnostics handlers usually expect inclusive ranges?

Reply:
I'm not sure about the other editors, but in the case of vc-code, yes, an inclusive range is expected.